addons/common/functions/fnc_fireWeapon.sqf

```diff
@@ -71,6 +71,10 @@ switch (true) do {
             _unit setAmmo [_muzzle, 1e6];
         };
 
+        if (_weapon isKindOf ["Laserdesignator_mounted", configFile >> "CfgWeapons"]) then {
```

Comment:
Is there a way to detect laser weapons in general (instead of relying on inheritance from this class)?

Reply:
We can check magazines https://github.com/zen-mod/ZEN/blob/5a4ccf6ddc0e78b079417f4992f0715c59927281/addons/common/script_component.hpp#L92

 or ammo https://github.com/zen-mod/ZEN/blob/bd5c43823cbb2dfe0f233cfedd1b4f2bdd254f63/addons/context_actions/functions/fnc_getVehicleWeaponActions.sqf#L18